### PR TITLE
Fix/date or stringy date

### PR DESCRIPTION
**Pull request by Noswad123**

### CODE REVIEW

File: test/compositionGuards.spec.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+	let date: Date;
+	beforeEach(function() {
+		date = new Date();
+	});
+	it('should return true for a date', function() {
+		assert.isTrue(isDateOrStringyDate(date));
+	});
+	it('should return true for a stringyDate date', function() {
+		assert.isTrue(isDateOrStringyDate(date.toISOString()));
+	});
```
Comment (shb7628) [inline test/compositionGuards.spec.ts]: negative cases?  what about an int?  what about a string that's not a date?

File: src/compositionGuards.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import {isString} from './primitiveGuards';
+import {isDate} from './instanceGuards';
+import {isDateString} from './parseGuards';
+import {thenGuard, orGuard} from './Combinators';
+
+export const isDateOrStringyDate = orGuard(
```
Comment (shb7628) [inline src/compositionGuards.ts]: if this is a public repo, not a fan of 'stringy', even thought that's the original code.
how about isDateOrStringDate . 
or isDateOrDateString

in fact, i wonder if you could just have an orGuard(isDateString, isDate);.  so many possibilities

File: package.json [hunk 3]
```diff
@@ -26,7 +26,10 @@
     "htmlcov:darwin": "open coverage/index.html",
     "clean": "rm -rf coverage/ dist/ .nyc_output/"
   },
-  "dependencies": {},
+  "dependencies": {
+    "@types/chance": "^1.0.4",
```
Comment (shb7628) [inline package.json]: are these dependencies or dev dependencies?

Comment (mgabeler-lee-6rs) [reply package.json]: This must be moved to be a dev dep... or just removed.  The tests don't need chance for what you're doing

File: src/compositionGuards.ts [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import {isString} from './primitiveGuards';
+import {isDate} from './instanceGuards';
+import {isDateString} from './parseGuards';
+import {thenGuard, orGuard} from './Combinators';
+
+export const isDateOrStringDate = orGuard(
```
Comment (mgabeler-lee-6rs) [inline src/compositionGuards.ts]: This guard and its test belongs in `restrictingGuards`

File: test/compositionGuards.spec.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+import * as Chance from 'chance';
+const chance = new Chance();
+import {isDateOrStringDate} from '../src';
+
+describe('compositionGuards', function() {
+	let date: Date;
+	beforeEach(function() {
+		date = new Date();
+	});
+	it('should return true for a date', function() {
```
Comment (mgabeler-lee-6rs) [inline test/compositionGuards.spec.ts]: Please look at the other tests in the repo for the standard way the guard tests are written.  There's a common pattern that should be kept.

File: test/compositionGuards.spec.ts (hunk 1 above)
Comment (mgabeler-lee-6rs) [reply test/compositionGuards.spec.ts]: agree, should have test cases for undefined, null, symbol, array, and object in addition to numbers

File: test/restrictingGuards.spec.ts [hunk 6]
```diff
@@ -2,6 +2,20 @@ import {assertGuards} from './assertGuards';
 import * as restricting from '../src/restrictingGuards';
 
 describe('restricting guards', function() {
+	context('isDateOrStringDate', function() {
+		const testSymbol = Symbol('test');
+		const badValues = [false, true, {datum: 1}, undefined, 6, 'string', testSymbol, null, [1, 2]];
+		const values = [new Date(), (new Date()).toISOString()];
+		const guard = restricting.isDateOrStringDate;
+
+		it('should return true for a date or date strings', function() {
```
Comment (mgabeler-lee-6rs) [inline test/restrictingGuards.spec.ts]:
```suggestion
		it('should guard for a date or date string', function() {
```

File: test/restrictingGuards.spec.ts [hunk 7]
```diff
@@ -2,6 +2,20 @@ import {assertGuards} from './assertGuards';
...
+			 values.forEach((value) => assertGuards(true)(guard, value));
+		});
+
+		it('should return false for a non-dates or none-date-strings', function() {
```
Comment (mgabeler-lee-6rs) [inline test/restrictingGuards.spec.ts]:
```suggestion
		it('should guard against non-dates and non-date-strings', function() {
```

File: package-lock.json [hunk 8]
```diff
@@ -327,6 +327,11 @@
       "integrity": "sha512-2Y8uPt0/jwjhQ6EiluT0XCri1Dbplr0ZxfFXUz+ye13gaqE8u5gL5ppao1JrUYr9cIip5S6MvQzBS7Kke7U9VA==",
       "dev": true
     },
+    "@types/chance": {
```
Comment (mgabeler-lee-6rs) [inline package-lock.json]: missed reverting changes to this file too

File: src/restrictingGuards.ts [hunk 9]
```diff
@@ -2,6 +2,16 @@ import {checkerToGuard} from './Checker';
 import {andGuard, orGuard, thenGuard} from './Combinators';
 import {ReasonGuard} from './ReasonGuard';
 import {isNumber, isSymbol, isString} from './primitiveGuards';
+import {isDate} from './instanceGuards';
+import {isDateString} from './parseGuards';
+
+export const isDateOrStringDate = orGuard(
```
Comment (mgabeler-lee-6rs) [inline src/restrictingGuards.ts]: Given the `parseGuards` bit is called `isDateString`, I think this should be called `isDateOrDateString` ... that also parses better as English IMO :)

**Pull request merged**